watershed_workflow/river_mesh.py

```diff
@@ -1181,3 +1181,72 @@ def adjust_hucs_for_river_corridor(hucs,
                                                                rc_points[rc_point_ind])
                     else:
                         rc_point_ind += 1
+
+
+def triangle_split_points(stream_triangles, river_corrs):
+    """Find additional points to split stream triangles.
+    
+    Parameters
+    ----------
+    stream_triangles : list of tuples
+        List of triangle vertices.
+    river_corrs : list of shapely.geometry.Polygon
+        List of river corridor polygons.
+    """
+    additional_points = []
+    for tri_verts in stream_triangles:
+        # Get midpoints of each edge
+        midpoints = [
```

Comment:
You could use watershed_workflow.utils.midpoint() here.  I've tried to prefer to use functions for this sort of thing, because:

1. it's easy to make typos and not see them and
2. calling a function called midpoint() is self-documenting -- you don't need comments like "get the midpoint of each edge"